src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -823,17 +837,29 @@ private boolean handleJmxNotificationForCassandra22(
             segmentId,
             repairNumber);
         failOutsideSynchronizedBlock = true;
```

Comment:
can we add an `assert` here that success hasn't already been marked true.

Reply:
Same as before, I don't see the point.